src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -67,24 +94,71 @@ class StdFunctionVectorCoefficient : public mfem::VectorCoefficient {
    * void(mfem::Vector &, mfem::Vector &). The first argument of the function is
    * the position, and the second argument is the output of the function.
    */
+
+  /**
+   * @brief Constructor that takes in a time-independent function of space that returns a vector
+   *
+   * @param[in] dim: number of components in the output vector
+   * @param[in] func: a function of space and time that returns a double
+   */
   StdFunctionVectorCoefficient(int dim, std::function<void(mfem::Vector&, mfem::Vector&)> func);
 
   /**
-   * @brief Evalate the coefficient at a quadrature point
+   * @brief Constructor that takes in a time-dependent function of space that returns a double
+   *
+   * @param[in] dim: number of components in the output vector
+   * @param[in] func: a function of space and time that returns a double
+   */
+  StdFunctionVectorCoefficient(int dim, std::function<void(mfem::Vector&, mfem::Vector&, double)> func);
+
+  /**
+   * @brief Evaluate the coefficient at a quadrature point
    *
    * @param[out] V The evaluated coefficient vector at the quadrature point
    * @param[in] T The element transformation for the evaluation
    * @param[in] ip The integration point for the evaluation
    */
   virtual void Eval(mfem::Vector& V, mfem::ElementTransformation& T, const mfem::IntegrationPoint& ip);
+  
+  bool is_time_dependent() const { return is_time_dependent_; } 
+
+  friend StdFunctionVectorCoefficient d_dt(const StdFunctionVectorCoefficient &, const double);
+  friend StdFunctionVectorCoefficient d2_dt2(const StdFunctionVectorCoefficient &, const double);
 
 private:
   /**
    * @brief The function to evaluate for the coefficient
    */
-  std::function<void(mfem::Vector&, mfem::Vector&)> func_;
+  std::function<void(mfem::Vector&, mfem::Vector&, double)> func_;
+  bool is_time_dependent_;
 };
 
+inline StdFunctionVectorCoefficient d_dt(const StdFunctionVectorCoefficient & y, const double dt = 1.0e-8) {
+  return StdFunctionVectorCoefficient(y.vdim, [dt, y = y.func_](mfem::Vector& x, mfem::Vector & return_value, double t) {
+    // if mfem would use return statements instead of 
+    // always using out parameters, we could just write
+    // return (y(x, t + dt) - y(x, t - dt)) / (2.0 * dt);
+
+    // instead, we get this:
+    mfem::Vector yl, yr;
+    y(x, yl, t - dt);
+    y(x, yr, t + dt);
+    return_value = (yr - yl) * (1.0 / (2.0 * dt));
+  });
+}
+
+inline StdFunctionVectorCoefficient d2_dt2(const StdFunctionVectorCoefficient & y, const double dt = 1.0e-4) {
```

Comment:
Again, I'm probably missing something but why is dt defaulted to 1.0e-4?

Reply:
like above, assumes the characteristic time is close to 1. The "sweet spot" for the second derivative appears to be at a larger value of dt than for the first derivative

![second_derivative_error](https://user-images.githubusercontent.com/16393433/94727396-85dc4780-0313-11eb-9191-0f46f83aa23d.png)

if the user isn't scaling their dimensions, they can provide their own appropriate value of dt.